### PR TITLE
update IRIS builder API for cleaner interface, client support

### PR DESCRIPTION
**Pull request by davebambrick**
### Background

Creating a separate gRPC endpoint for each seed data type is overly cumbersome, since, barring [de]serialization steps, the data flows identically to the underlying `IrisBuilder`. Instead, we can leverage Protobuf's `oneof` feature ([Language guide](https://protobuf.dev/programming-guides/proto3/#oneof),  [C++ code guide](https://protobuf.dev/reference/cpp/cpp-generated/#oneof-numeric)) to use the same request message to support all our present (and future) seed data formats.

### What's new

- [x] Remove `StartBuildFromConfsRequest`, `StartBuildFromEdgesRequest` in favor of a single `StartBuildRequest` which can take at most one of a set of types of seed data.
- [x] Add `id` field to `PlanContext` message. This enables a user to start a job for an already-extant context without resending all the data.
- [x] Remove `iris_client.py` example (moved into `planning_service` proper).
- [x] Naming refactor for uniformity with convention in `planning_service` (e.g., referring to edges as `(u, v)` vs `(v1, v2)`).

### Related work

- [ ] [link] needs this PR

### TODOs / Nice-To-Haves

- [ ] [Add system tests for new feature.]

### CODE REVIEW

File: proto/planning/basic_types.proto [hunk 1]
```diff
@@ -24,10 +24,11 @@ message SystemConf {
 // space.
 message SystemConfEdge {
   // Vertex 1
-  SystemConf v1 = 1;
+  SystemConf u = 1;
```
Comment (davebambrick) [inline proto/planning/basic_types.proto:27]: to match internal naming elsewhere.

File: proto/planning/basic_types.proto [hunk 2]
```diff
@@ -24,10 +24,11 @@ message SystemConf {
...
   // Vertex 2
-  SystemConf v2 = 2;
+  SystemConf v = 2;
```
Comment (davebambrick) [inline proto/planning/basic_types.proto:29]: to match internal naming elsewhere.

File: proto/planning/basic_types.proto [hunk 3]
```diff
@@ -95,7 +96,7 @@ message AngleBetweenVectorsConstraint {
   double angle_upper = 6;
 }
 // The set of all constraints for a given planning problem.
-message Constraints {
+message ConstraintsSet {
```
Comment (davebambrick) [inline proto/planning/basic_types.proto:99]: Renamed due to ambiguity vs. individual constraint types.

File: proto/planning/basic_types.proto [hunk 4]
```diff
@@ -177,15 +178,17 @@ message PlanContextId {
...
-//   3.) A set of Constraints, which impose certain geometric constraints on the
-//   system.
+//   3.) A set of ConstraintsSet, which impose certain geometric constraints on
+//   the system.
 message PlanContext {
   // The associated top-level system name.
   string system_name = 1;
   // The model directive YAML contents as string.
-  string model_directive_yaml = 3;
+  string model_directive_raw = 2;
```
Comment (davebambrick) [inline proto/planning/basic_types.proto:187]: Renamed to match internal convention.

**Review by sadraddini — COMMENTED** [review]
LGTM

**Pull request merged**